lib/liquid/combinators/tags/for.ex

```diff
@@ -51,6 +51,21 @@ defmodule Liquid.Combinators.Tags.For do
   alias Liquid.Combinators.{General, Tag}
   alias Liquid.Combinators.Tags.Generic
 
+  @type t :: [
+          for: [
+            for_statements: [
+              variable: String.t(),
+              value: LexicalToken.value(),
+              for_params: [
+                offset: Integer.t() | String.t(),
+                limit: Integer.t() | String.t()
+              ],
+              for_body: Liquid.NimbleParsec.t(),
+              else: Generic.t()
```

Comment:
Generic is not a type, is a helper (same NimbleParser). You can create `Generic.else()` type and use it directly: `for_body: Liquid.t() | Generic.else()` please note `else` is optional